clients/src/main/java/org/apache/kafka/common/requests/ProduceResponse.java

```diff
@@ -149,9 +155,29 @@
      */
     private static final Schema PRODUCE_RESPONSE_V7 = PRODUCE_RESPONSE_V6;
 
+    /**
+     * V8 adds error_records and error_message. (see KIP-467)
+     */
+    public static final Schema PRODUCE_RESPONSE_V8 = new Schema(
+            new Field(RESPONSES_KEY_NAME, new ArrayOf(new Schema(
+                    TOPIC_NAME,
+                    new Field(PARTITION_RESPONSES_KEY_NAME, new ArrayOf(new Schema(
+                            PARTITION_ID,
+                            ERROR_CODE,
+                            new Field(BASE_OFFSET_KEY_NAME, INT64),
+                            new Field(LOG_APPEND_TIME_KEY_NAME, INT64, "The timestamp returned by broker after appending " +
+                                    "the messages. If CreateTime is used for the topic, the timestamp will be -1. " +
+                                    "If LogAppendTime is used for the topic, the timestamp will be the broker local " +
+                                    "time when the messages are appended."),
+                            LOG_START_OFFSET_FIELD,
+                            new Field(ERROR_RECORDS_KEY_NAME, new ArrayOf(INT32), "The relative offset of records" +
+                                    "that cause the batch to be dropped"),
+                            ERROR_MESSAGE_FIELD)))))),
```

Comment:
@guozhangwang Is it useful to have only a single error message? There may be multiple failures?

Reply:
We would only return one type of errors: if it is due to record validation which could happen on multiple records within a batch, then we would use that error message with the list of relative offsets; for other errors like magic byte mismatch etc we will return upon the first occurrence and the list of relative offsets will be a singleton.

If multiple of such different errors happen, we would still only return one error message (the second category would take precedence).